### PR TITLE
fix(rag): knowledge base searche error

### PR DESCRIPTION
**Pull request by jiyegege**
### What problem does this PR solve?

1. Fix BeartypeCallHintParamViolation Error for Knowledge Base Searches
Pasted Graphic 7

```
Traceback (most recent call last):
  File "ragflow/api/apps/chunk_app.py", line 286, in retrieval_test
    ranks = retr.retrieval(question, embd_mdl, kb.tenant_id, kb_ids, page, size,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "ragflow/rag/nlp/search.py", line 304, in retrieval
    sres = self.search(req, [index_name(tid) for tid in tenant_ids], kb_ids, embd_mdl, highlight)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<@beartype(rag.nlp.search.Dealer.search) at 0x3375ea2a0>", line 72, in search
  File "ragflow/rag/nlp/search.py", line 88, in search
    matchText, keywords = self.qryr.question(qst, min_match=0.3)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<@beartype(rag.nlp.query.FulltextQueryer.question) at 0x3375e9760>", line 31, in question
  File "ragflow/rag/nlp/query.py", line 180, in question
    return MatchTextExpr(
           ^^^^^^^^^^^^^^
  File "<@beartype(rag.utils.doc_store_conn.MatchTextExpr.__init__) at 0x152abdb20>", line 28, in __init__
beartype.roar.BeartypeCallHintParamViolation: Method rag.utils.doc_store_conn.MatchTextExpr.__init__() parameter fields=['title_tks^10', 'title_sm_tks^5', 'important_kwd^30', 'important_tks^20', 'content_ltks^2'...'] violates type hint <class 'str'>, as list ['title_tks^10', 'title_sm_tks^5', 'important_kwd^30', 'important_tks^20', 'content_ltks^2'...'] not instance of str.
```

In line 181 of query.py, self.query_fields is of type list, but MatchTextExpr needs to initialize arguments of type str
￼
Pasted Graphic 8
inheritance

￼
Looking for places where the "fields" field is used, I found that it is always being used as a list, so the "fields" type can be changed to list[str].
￼
Pasted Graphic 10

Alright, now everything is working properly.
￼
Pasted Graphic 11

### Type of change

- [X] Bug Fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: rag/utils/doc_store_conn.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __repr__(self):
 class MatchTextExpr(ABC):
     def __init__(
         self,
-        fields: str,
+        fields: list[str],
```
Comment (KevinHuSh) [inline rag/utils/doc_store_conn.py]:
```suggestion
        fields: str|list[str],
```

**Comment by yuzhichang** [comment]
@jiyegege I've fixed the issue by changing `MatchTextExpr.fields' to `list[str]`.
Thanks for your fix anyway!

**Comment by dream-1ab** [comment]
how can I use this fix immediately in my local computer please? should I rebuild docker container first?

**Comment by dream-1ab** [comment]
```
 > build
148.3 > umi build
148.3 
150.2 info  - [你知道吗？] 如果你有 MPA（多页应用）需求，可尝试新出的 mpa 配置项，详见 https://umijs.org/docs/guides/mpa
150.2 info  - Umi v4.3.34
151.7 info  - Preparing...
153.3 info  - [icons] generate icons local:google, local:github
155.0 
155.0 Rebuilding...
156.6 
156.6 Done in 1731ms.
156.6 info  - [plugin: ./node_modules/@umijs/plugins/dist/tailwindcss] tailwindcss service exited
158.9 ℹ Compiling Webpack
194.9 [BABEL] Note: The code generator has deoptimised the styling of /ragflow/web/src/assets/svg/login-avatars.svg as it exceeds the max of 500KB.
541.7 
541.7 <--- Last few GCs --->
541.7 
541.7 [198:0x59f6000]   389807 ms: Mark-Compact 966.2 (997.5) -> 965.4 (996.8) MB, 1568.38 / 0.00 ms  (average mu = 0.108, current mu = 0.039) allocation failure; scavenge might not succeed
541.7 [198:0x59f6000]   391999 ms: Mark-Compact (reduce) 975.3 (998.8) -> 974.2 (999.0) MB, 1483.38 / 0.01 ms  (+ 446.0 ms in 91 steps since start of marking, biggest step 22.8 ms, walltime since start of marking 1968 ms) (average mu = 0.115, current mu = 0.120
541.7 
541.7 <--- JS stacktrace --->
541.7 
541.7 FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
541.7 ----- Native stack trace -----
541.7 
541.7  1: 0xb8ced1 node::OOMErrorHandler(char const*, v8::OOMDetails const&) [umi]
541.7  2: 0xf06460 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [umi]
541.7  3: 0xf06747 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [umi]
541.8  4: 0x11182e5  [umi]
541.8  5: 0x1118874 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [umi]
541.8  6: 0x112f764 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::internal::GarbageCollectionReason, char const*) [umi]
541.8  7: 0x112ff7c v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [umi]
541.8  8: 0x1106281 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [umi]
541.8  9: 0x1107415 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [umi]
541.8 10: 0x10e4a66 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [umi]
541.8 11: 0x1540896 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [umi]
541.8 12: 0x1979ef6  [umi]
542.2 Aborted (core dumped)

```

**Comment by dream-1ab** [comment]
Why does the front-end need to rebuild webpack? Can't we use the existing webpack?

**Comment by dream-1ab** [comment]
> docker build process here...
> 
> `=> [builder 4/6] RUN --mount=type=cache,id=ragflow_builder_npm,target=/root/.npm,sharing=locked cd web && npm install --force && npm run build `
> 
> ```
>  > build
> 148.3 > umi build
> 148.3 
> 150.2 info  - [你知道吗？] 如果你有 MPA（多页应用）需求，可尝试新出的 mpa 配置项，详见 https://umijs.org/docs/guides/mpa
> 150.2 info  - Umi v4.3.34
> 151.7 info  - Preparing...
> 153.3 info  - [icons] generate icons local:google, local:github
> 155.0 
> 155.0 Rebuilding...
> 156.6 
> 156.6 Done in 1731ms.
> 156.6 info  - [plugin: ./node_modules/@umijs/plugins/dist/tailwindcss] tailwindcss service exited
> 158.9 ℹ Compiling Webpack
> 194.9 [BABEL] Note: The code generator has deoptimised the styling of /ragflow/web/src/assets/svg/login-avatars.svg as it exceeds the max of 500KB.
> 541.7 
> 541.7 <--- Last few GCs --->
> 541.7 
> 541.7 [198:0x59f6000]   389807 ms: Mark-Compact 966.2 (997.5) -> 965.4 (996.8) MB, 1568.38 / 0.00 ms  (average mu = 0.108, current mu = 0.039) allocation failure; scavenge might not succeed
> 541.7 [198:0x59f6000]   391999 ms: Mark-Compact (reduce) 975.3 (998.8) -> 974.2 (999.0) MB, 1483.38 / 0.01 ms  (+ 446.0 ms in 91 steps since start of marking, biggest step 22.8 ms, walltime since start of marking 1968 ms) (average mu = 0.115, current mu = 0.120
> 541.7 
> 541.7 <--- JS stacktrace --->
> 541.7 
> 541.7 FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
> 541.7 ----- Native stack trace -----
> 541.7 
> 541.7  1: 0xb8ced1 node::OOMErrorHandler(char const*, v8::OOMDetails const&) [umi]
> 541.7  2: 0xf06460 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [umi]
> 541.7  3: 0xf06747 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [umi]
> 541.8  4: 0x11182e5  [umi]
> 541.8  5: 0x1118874 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [umi]
> 541.8  6: 0x112f764 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::internal::GarbageCollectionReason, char const*) [umi]
> 541.8  7: 0x112ff7c v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [umi]
> 541.8  8: 0x1106281 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [umi]
> 541.8  9: 0x1107415 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [umi]
> 541.8 10: 0x10e4a66 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [umi]
> 541.8 11: 0x1540896 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [umi]
> 541.8 12: 0x1979ef6  [umi]
> 542.2 Aborted (core dumped)
> ```

this issue goes off after `export NODE_OPTIONS="--max-old-space-size=8192` so the RUN command in Dockerfile.slim becomes `RUN --mount=type=cache,id=ragflow_builder_npm,target=/root/.npm,sharing=locked cd web && export NODE_OPTIONS="--max-old-space-size=8192" && npm install --force && npm run build`

**Pull request closed without merging**